sdks/python/apache_beam/io/jdbc.py

```diff
@@ -168,41 +182,33 @@ def __init__(
     :param expansion_service: The address (host:port) of the ExpansionService.
     """
 
-    super(WriteToJdbc, self).__init__(
-        self.URN,
-        NamedTupleBasedPayloadBuilder(
-            WriteToJdbcSchema(
-                driver_class_name=driver_class_name,
-                jdbc_url=jdbc_url,
-                username=username,
-                password=password,
-                statement=statement,
-                connection_properties=connection_properties,
-                connection_init_sqls=connection_init_sqls,
+        super(WriteToJdbc, self).__init__(
+            self.URN,
+            NamedTupleBasedPayloadBuilder(
+                ReadFromWriteToJdbcSchema(
+                    location=jdbc_url,
```

Comment:
I think the better analog for "location" would actually be the table name, and the JDBC URL would be part of the configuration. Unfortunately it looks like the typical use-case (i.e. what we have in xlang_jdbcio_it_test) just has the table name implicitly in the query: https://github.com/apache/beam/blob/84cbd9bdfbe45221b8f3302cedad7bdb47cd2f5a/sdks/python/apache_beam/io/external/xlang_jdbcio_it_test.py#L151

and in the statement:
https://github.com/apache/beam/blob/84cbd9bdfbe45221b8f3302cedad7bdb47cd2f5a/sdks/python/apache_beam/io/external/xlang_jdbcio_it_test.py#L120

Maybe what we should do here is require a `table_name` parameter that we will use for the location. Then, in the JdbcSchemaIOProvider, we can actually generate the query and statement. The user would still have the ability to override both of these if they prefer. WDYT?

Reply:
Also for the shared configuration object we should disambiguate query and statement, maybe as read_query and write_statement. The parameters for ReadFromJdbc and WriteToJdbc could remain "query" and "statement" though since there it's clear from context.

Reply:
That makes sense to me. I added in the query/statement generator code and modified the existing test to use the default generated query/statement, but I think we should add in overriding query/statement tests too.